prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -659,6 +660,27 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
             ?.getValue(DSL.max(SETTING.CREATED_AT))
     }
 
+    /**
+     * Get the number of action records with action_params = [queueMessage]
+     */
+    fun getActionCountForReport(queueMessage: String): Int {
+        val oneDayAgo = OffsetDateTime.now().minusDays(1)
+        return create
+            .select(inline(1))
+            .from(ACTION)
+            // cheap way to limit the number of actions that need to be looked at - probably don't even need to look
+            //  this far back but this should be called rarely if ever in any case since it is only when a 'process'
+            //  fails and needs to be retried
+            .where(ACTION.CREATED_AT.greaterOrEqual(oneDayAgo))
+            .and(
+                ACTION.ACTION_NAME.eq(TaskAction.process)
+                    .or(ACTION.ACTION_NAME.eq(TaskAction.process_warning))
+                    .or(ACTION.ACTION_NAME.eq(TaskAction.process_error))
+            )
+            .and(ACTION.ACTION_PARAMS.eq(queueMessage))
+            .count()
+    }
```

Comment:
So there's a bit of an off-by-one error here.

Mine prints:

```
    1043 | process_warning | process&PROCESS&4acb8667-afdf-4d5a-b305-17ba448ddd7f&None&&& | Failed to process 0 times, setting status to process_warning. | 2021-12-16 17:26:34.617375+00 |                 |             |                |           |             |                    |               | 
      1044 | process_warning | process&PROCESS&4acb8667-afdf-4d5a-b305-17ba448ddd7f&None&&& | Failed to process 1 times, setting status to process_warning. | 2021-12-16 17:26:36.001552+00 |                 |             |                |           |             |                    |               | 
      1045 | process_warning | process&PROCESS&4acb8667-afdf-4d5a-b305-17ba448ddd7f&None&&& | Failed to process 2 times, setting status to process_warning. | 2021-12-16 17:26:36.606063+00 |                 |             |                |           |             |                    |               | 
      1046 | process_warning | process&PROCESS&4acb8667-afdf-4d5a-b305-17ba448ddd7f&None&&& | Failed to process 3 times, setting status to process_warning. | 2021-12-16 17:26:36.717908+00 |                 |             |                |           |             |                    |               | 
      1047 | process_error   | process&PROCESS&4acb8667-afdf-4d5a-b305-17ba448ddd7f&None&&& | Failed to process 4 times, setting status to process_error.   | 2021-12-16 17:26:36.829744+00 |                 |             |                |           |             |                    |               | 

```

Note the "0" in Failed to process 0 times.  ;) 

I think this is because the Action hasn't been written to the database at the time the query is happening.

Reply:
Fixed